DataSetExplorer/DataSetBuilder/FullDataSetBuilderService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using CodeModel;
+using CodeModel.CaDETModel;
+using DataSetExplorer.DataSetBuilder.Model;
+using DataSetExplorer.DataSetSerializer;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+
+namespace DataSetExplorer
+{
+    class FullDataSetBuilderService : IFullDataSetBuilder
```

Comment:
We should also drop the Service keyword because this is a domain service, to avoid confusion with application services. Actually, we don't need the interface, as the domain service is part of the core layer (in the same way we don't need interfaces for entities and value objects).

We should drop the Builder keyword, as that signals the "Builder" pattern, and this class is not that. I would rename it to FullDataSetFactory, as it's closer to that pattern.